src/Workspaces/Core/Portable/Utilities/BKTree.Serialization.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Roslyn.Utilities
```

Comment:
/cc: @pilchie - I think I recall a conversation about file names where there was a distinction between the punctuation used for "Type.NestedType" and "Type_PartOfTheTypesBehavior". Is that something we want to do?